runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -150,14 +157,19 @@ public static int attachXWalkCore(Context context) {
         Log.d(TAG, "Attach xwalk core");
         sProvisionalInstance = new XWalkCoreWrapper(context, -1);
         if (!sProvisionalInstance.findEmbeddedCore()) {
-            int status = sProvisionalInstance.mCoreStatus;
+            if(sProvisionalInstance.isDownloadMode()) {
```

Comment:
There's no need to differentiate download mode. Check embedded mode -> download mode -> shared mode in sequence.